hadoop/src/test/java/org/apache/metamodel/util/HdfsResourceIntegrationTest.java

```diff
@@ -1,187 +1,244 @@
-
-        res1.getHadoopFileSystem().delete(res1.getHadoopPath(), false);
-        logger.info(stopwatch.elapsed(TimeUnit.MILLISECONDS) + " - deleted");
-
-        Assert.assertFalse(res1.isExists());
-
-        logger.info(stopwatch.elapsed(TimeUnit.MILLISECONDS) + " - done");
-        stopwatch.stop();
-    }
-
-}
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.metamodel.util;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Stopwatch;
+
+public class HdfsResourceIntegrationTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(HdfsResourceIntegrationTest.class);
+
+    private String _filePath;
+    private String _hostname;
+    private int _port;
+
+    @Before
+    public void setUp() throws Exception {
+        final File file = new File(getPropertyFilePath());
+        final boolean configured;
+        if (file.exists()) {
+            final Properties properties = new Properties();
+            properties.load(new FileReader(file));
+            _filePath = properties.getProperty("hadoop.hdfs.file.path");
+            _hostname = properties.getProperty("hadoop.hdfs.hostname");
+            final String portString = properties.getProperty("hadoop.hdfs.port");
+            configured = _filePath != null && _hostname != null && portString != null;
+            if (configured) {
+                _port = Integer.parseInt(portString);
+            }
+        } else {
+            configured = false;
+        }
+        Assume.assumeTrue(configured);
+    }
+
+    @After
+    public void tearDown() throws Exception {
```

Comment:
Diff not visible, so marking the place where changes are. Added a tearDown method to make sure we don't leave our test file in the cluster.